src/Regex.cpp

```diff
@@ -25,6 +25,19 @@ boost::shared_ptr<RegexMatch> Regex::find(const std::string &expression) const {
     return boost::make_shared<FindRegexMatch>(regexImpl, expression);
 }
 
+namespace {
+bool isStartOfCodepoint(unsigned int i) {
+    return i < 0x80 || (i & 0xC0) == 0xC0;
+}
+
+std::ptrdiff_t codepointOffset(const std::string& expression, size_t bytePosition) {
```

Comment:
Please standardise on `ptrdiff_t`. Especially because you cast from `std::ptrdiff_t` to `std::size_t` at the call site.

Also 🐦: This code assumes that `expression` is UTF-8 encoded and that characters are 8-bit values. Both of these hold, but I don't see why you talk about bytes here. Technically it's correct, but we know much more about these "bytes": They are UTF-8 code units. This is much more specific than "bytes". This also applies to the commit message.